### PR TITLE
contract changes to gmsa linux support

### PR DESCRIPTION
**Pull request by saikiranakula-amzn**
Summary
Updated  client contracts to communicate with credentials-fetcher to support gMSA Linux

Implementation details
The changes are to add the client support to have integration with new contracts exposed by credentials-fetcher

Testing
Added unit tests both positive and negative test case for the functionality. Also manually tested ensuring ECS gMSA works as expected

New tests cover the changes:

Description for the changelog
Updated contracts to credentials-fetcher client to support gMSA linux

The new contracts accept the credentialspec s3 arn input, and the associated task execution role credentials to process the request and generate associated kerberos tickets

Does this PR include breaking model changes? If so, Have you added transformation functions?

Licensing
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: ecs-agent/gmsacredclient/credentialsfetcherclient.go [hunk 1]
```diff
@@ -37,13 +42,7 @@ func GetGrpcClientConnection() (*grpc.ClientConn, error) {
 
 // getSocketAddress() returns the credentials-fetcher socket dir
 func getSocketAddress() (string, error) {
```
Comment (amogh09) [inline ecs-agent/gmsacredclient/credentialsfetcherclient.go:44]: Do we still need the `error` in function signature?

File: ecs-agent/gmsacredclient/credentialsfetcherclient.go [hunk 2]
```diff
@@ -18,6 +17,12 @@ type CredentialsFetcherClient struct {
 	timeout time.Duration
 }
 
+const (
+	// credentialsFetcherSocketLocation is the path to unix domain socket for
+	// communicating with credentials-fetcher daemon.
+	credentialsFetcherSocketLocation = "/var/credentials-fetcher/socket/credentials_fetcher.sock"
```
Comment (amogh09) [inline ecs-agent/gmsacredclient/credentialsfetcherclient.go:23]: Would this be an issue on Windows?

Comment (saikiranakula-amzn) [reply ecs-agent/gmsacredclient/credentialsfetcherclient.go:23]: credentialsfetcher is only linux, there won't be any issue with windows

File: ecs-agent/gmsacredclient/credentialsfetcherclient.go (hunk 1 above)
Comment (saikiranakula-amzn) [reply ecs-agent/gmsacredclient/credentialsfetcherclient.go]: we need the method back again

**Review by amogh09 — APPROVED** [review]

**Pull request merged**